File: API/course_project_spring_2023_api/Models/Tag.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace course_project_spring_2023_api.Models
{
    public class Tag
    {
        [Required]
        public int Id { get; set; }
        
        [Required]
        public string Name { get; set; } = "default";

        [Required]
        [JsonIgnore]
        public virtual IList<Blog> Blogs { get; set; } = new List<Blog>();
    }
}

File: APP/course_project_spring_2023_app/packages/api_repository/lib/src/models/tag.dart
class Tag {
  const Tag({required this.id, required this.name});

  final int id;
  final String name;

  factory Tag.fromJson(Map<String, dynamic> json) {
    return Tag(id: json['Id'], name: json['Name']);
  }

  Map<String, dynamic> toJson() {
    return <String, dynamic>{
      "Name": name,
    };
  }

  @override
  String toString() {
    return "Id: $id, Name: $name";
  }
}
